internal/configs/module_call.go

```diff
@@ -72,7 +72,10 @@ func decodeModuleBlock(block *hcl.Block, override bool) (*ModuleCall, hcl.Diagno
 	}
 
 	if attr, exists := content.Attributes["version"]; exists {
-		mc.VersionAttr = attr
+		val, _ := attr.Expr.Value(nil)
```

Comment:
Unfortunately, this does not work with static (early) evaluation as requested in the issue.  Take a look at the decodeStaticVersion function further down in this file.